lib/services/devices/registrationUtils.js

```diff
@@ -299,6 +297,62 @@ function sendRegistrationsNgsi2(unregister, deviceData, callback) {
     utils.executeWithSecurity(options, deviceData, createRegistrationHandlerNgsi2(unregister, deviceData, callback));
 }
 
+function sendUnsubscriptionsNgsi2(deviceData, callback) {
+    if (deviceData.subscriptionId) {
+        logger.debug(
+            context,
+            'Sending v2 device unsubscriptions to Context Broker at [%s]',
+            config.getConfig().contextBroker.url
+        );
+        logger.debug(context, 'Using the following subscriptionId %j', deviceData.subscriptionId);
+        subscriptionService.unsubscribe(deviceData, deviceData.subscriptionId, callback);
+    }
+    logger.debug(context, 'No subscription found for unregister');
+    return callback(null, deviceData);
+}
+
+function sendSubscriptionsNgsi2(unregister, deviceData, callback) {
+    function updateSubscriptionNgsi2(deviceData, callback) {
+        const functions = [];
+
+        function removeSubscriptionId(deviceData, unregistrationResult, callback) {
+            delete deviceData.subscriptionId;
+            return callback(null, deviceData);
+        }
+
+        functions.push(async.apply(sendSubscriptionsNgsi2, true, deviceData));
+        functions.push(async.apply(removeSubscriptionId, deviceData));
+        functions.push(async.apply(sendSubscriptionsNgsi2, false));
+        async.waterfall(functions, callback);
+    }
+
+    if (unregister) {
+        return sendUnsubscriptionsNgsi2(deviceData, callback);
+    }
+    if (deviceData.subscriptionId) {
+        return updateSubscriptionNgsi2(deviceData, callback);
+    }
+    const attrs = []
+        .concat(formatAttributes(deviceData.lazy), formatAttributes(deviceData.commands))
+        .reduce(mergeWithSameName, []);
+
+    if (attrs.length === 0) {
+        logger.debug(context, 'Subscription is not needed. Device without lazy attrs or commands');
```

Comment:
Not sure if the references to "lazy attrs" is correct here... The subscription will only cover the commands case, not lazy attributes.